### PR TITLE
fix(planstate): don't hold planLock while calling plan-changed listeners

### PR DESCRIPTION
**Pull request by benhoyt**
Some listeners do quite a bit of stuff (check manager and log manager), and they're defined in packages external to the planstate package, so it's best to avoid holding our plan lock for the duration.

See [spec OP040](https://docs.google.com/document/d/1cmZ5FWz0efglDAxGRciY5GRyD6KfWJCOe-kjWl6auig/edit#heading=h.po9vomna3y7d).

Also:

- PlanManager doesn't use state and runner, so remove them
- Rename planHandlers to changeListeners to match AddChangeListener

### CODE REVIEW

**Review by IronCore864 — APPROVED** [review]

File: internals/overlord/planstate/manager.go [hunk 1]
```diff
@@ -33,23 +32,19 @@ func (e *LabelExists) Error() string {
 }
 
 type PlanManager struct {
-	state     *state.State
-	runner    *state.TaskRunner
```
Comment (benhoyt) [inline internals/overlord/planstate/manager.go:37]: Drive-by: PlanManager never used state and runner (maybe a copy 'n' paste issue), so remove them.

File: internals/overlord/planstate/manager.go [hunk 2]
```diff
@@ -76,14 +74,16 @@ type PlanChangedFunc func(p *plan.Plan)
...
 
-func (m *PlanManager) planChanged(plan *plan.Plan) {
-	m.plan = plan
-	for _, f := range m.planHandlers {
+func (m *PlanManager) callChangeListeners(plan *plan.Plan) {
+	if plan == nil {
+		// Avoids if statement on every deferred call to this method (we
+		// shouldn't call listeners when the operation fails).
+		return
+	}
```
Comment (hpidcock) [inline internals/overlord/planstate/manager.go:85]: If we also had a `recover()` higher up, we'd want to recover here then re-panic rather than call the changeListeners. But I don't think we recover anywhere do we?

**Review by hpidcock — APPROVED** [review]
Thanks LGTM. Just the one question.

File: internals/overlord/planstate/manager.go (hunk 2 above)
Comment (benhoyt) [reply internals/overlord/planstate/manager.go:85]: We do recover and re-panic in Run(). However, in case of a panic in the functions that call `callChangeListeners`, plan (newPlan) will be nil, so it won't call them due to this `if` statement.
